modules/ensemble/lib/widget/collapsible.dart

```diff
@@ -0,0 +1,345 @@
+      'rightIcon': (value) => _controller.rightIcon = value,
+      'flipLeftIconIfOpen': (value) =>
+          _controller.flipLeftIconIfOpen = Utils.optionalBool(value),
+      'flipRightIconIfOpen': (value) =>
+          _controller.flipRightIconIfOpen = Utils.optionalBool(value),
+      'paddingListTop': (value) =>
+          _controller.paddingListTop = Utils.getDouble(value, fallback: 0.0),
+      'paddingListBottom': (value) =>
+          _controller.paddingListBottom = Utils.getDouble(value, fallback: 0.0),
+      'paddingListHorizontal': (value) => _controller.paddingListHorizontal =
+          Utils.getDouble(value, fallback: 0.0),
+      'paddingBetweenOpenSections': (value) =>
+          _controller.paddingBetweenOpenSections = Utils.optionalDouble(value),
+      'paddingBetweenClosedSections': (value) => _controller
+          .paddingBetweenClosedSections = Utils.optionalDouble(value),
+      'disableScrolling': (value) =>
+          _controller.disableScrolling = Utils.getBool(value, fallback: false),
+      'openAndCloseAnimation': (value) =>
+          _controller.openAndCloseAnimation = Utils.optionalBool(value),
+      'scaleWhenAnimating': (value) =>
+          _controller.scaleWhenAnimating = Utils.optionalBool(value),
+      'accordionId': (value) =>
+          _controller.accordionId = Utils.optionalString(value),
+    };
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {
+      'openSection': (int index) => _controller.openSection(index),
+      'closeSection': (int index) => _controller.closeSection(index),
+    };
+  }
+}
+
+class CollapsibleController extends BoxController {
+  List<YamlMap> items = [];
+  bool isAccordion = true;
+  int? initialOpeningSequenceDelay;
+  HeaderStyle headerStyle = HeaderStyle();
+  ContentStyle contentStyle = ContentStyle();
+  dynamic leftIcon;
+  dynamic rightIcon;
+  bool? flipLeftIconIfOpen;
+  bool? flipRightIconIfOpen;
+  double paddingListTop = 0.0;
+  double paddingListBottom = 0.0;
+  double paddingListHorizontal = 0.0;
+  double? paddingBetweenOpenSections;
+  double? paddingBetweenClosedSections;
+  bool disableScrolling = false;
+  bool? openAndCloseAnimation;
+  bool? scaleWhenAnimating;
+  String? accordionId;
+
+  final ValueNotifier<List<int>> _openSections = ValueNotifier<List<int>>([]);
+  List<int> get openSections => _openSections.value;
+
+  void openSection(int index) {
+    if (isAccordion) {
+      _openSections.value = [index];
+    } else {
+      final newOpenSections = List<int>.from(_openSections.value);
+      if (newOpenSections.contains(index)) {
+        newOpenSections.remove(index);
+      } else {
+        newOpenSections.add(index);
+      }
+      _openSections.value = newOpenSections;
+    }
+  }
+
+  void closeSection(int index) {
+    final newOpenSections = List<int>.from(_openSections.value);
+    newOpenSections.remove(index);
+    _openSections.value = newOpenSections;
+  }
+
+  @override
+  void dispose() {
```

Comment:
This is wrong because you don't call super.dispose(). Remove the function if you don't use it.

Reply:
Fixed @vusters